core/src/main/java/org/elasticsearch/search/rescore/RescoreParseElement.java

```diff
@@ -71,7 +74,7 @@ public void parseSingleRescoreContext(XContentParser parser, SearchContext conte
         if (windowSize != null) {
             rescoreContext.setWindowSize(windowSize.intValue());
         }
-        context.addRescore(rescoreContext);
+        return rescoreContext;
```

Comment:
This enables us to do roundtrip-tests for the new `build()` method as long as this class is not gone. It's the same approach I took with the highlighters. The utility method `parseSingleRescoreContext` now returns a RescoreSearchContext rather than requiring a SearchContext to set it on. I think this change is small enough given that it enables us to check that we build the same RescoreSearchContext objects like before in the tests.